### PR TITLE
INT: use template to fill unknown field types in Create(Tuple)StructIntention

### PR DESCRIPTION
**Pull request by Kobzol**
This PR adds a template builder to fill in unknown types in `CreateStructIntention` and `CreateTupleStructIntention`.

Structs:
struct

Tuple structs:
tuple-struct

It seems that the hacky `RsMultipleVariableRenamer` is no longer needed with the new template API, I'll try to remove it in another PR.

Fixes: https://github.com/intellij-rust/intellij-rust/issues/7672

changelog: Intentions to create structs and tuple structs will now let the user fill in unknown field types.

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/intentions/createFromUsage/CreateStructIntention.kt [hunk 1]
```diff
@@ -50,9 +48,23 @@ class CreateStructIntention : RsElementBaseIntentionAction<CreateStructIntention
 
         val fields = inserted.blockFields?.namedFieldDeclList.orEmpty()
         if (inserted.containingFile == function.containingFile && fields.isNotEmpty()) {
-            editor.caretModel.moveToOffset(fields[0].textOffset)
-            RsMultipleVariableRenamer(fields.map { it.createSmartPointer() })
-                .doRename(fields[0], editor, DataManager.getInstance().getDataContext(editor.component))
+            val unknownTypes = inserted.descendantsOfType<RsBaseType>().filter { it.underscore != null }
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/intentions/createFromUsage/CreateStructIntention.kt:51]: Let's map them to smart pointers, and `fields` as well. Raw PSI may not outlive `newTemplateBuilder` call.

**Review by vlad20012 — APPROVED** [review]

File: src/main/kotlin/org/rust/ide/intentions/createFromUsage/CreateStructIntention.kt (hunk 1 above)
Comment (Kobzol) [reply src/main/kotlin/org/rust/ide/intentions/createFromUsage/CreateStructIntention.kt:57]: Changed to smart pointers. I was on a vacation, thus the delay.

**Comment by Kobzol** [comment]
bors r=vlad20012

**Comment by Kobzol** [comment]
bors retry

**Pull request merged**